packages/edit-widgets/src/store/transformers.js

```diff
@@ -3,13 +3,26 @@
  */
 import { createBlock, parse, serialize } from '@wordpress/blocks';
 
+function __experimentalAddWidgetIdToBlock( block, widgetId ) {
+	return {
+		...block,
+		attributes: {
+			...( block.attributes || {} ),
+			__internalWidgetId: widgetId,
```

Comment:
It's a bit weird that we add an attribute that is not registered. I'm fine with it but I hope it's something that can break if for some reason, in the future, we apply the same "sanitizeAttributes" behavior in flows where it's not applied right now.

Let's make sure we've tests to cover this.

Reply:
> I'm fine with it but I hope it's something that can break if for some reason, in the future, we apply the same "sanitizeAttributes" behavior in flows where it's not applied right now.

I'm not sure what you meant by that? What kind of tests you have in mind?

Reply:
Not particular tests in my mind, this is meant to fix widgets updating if I'm not wrong, let's just make sure it's covered because it can easily break.

Reply:
Got it! Just added a test in af4156b and reverted the hack introduced in #28078.